torch_geometric/nn/aggr/basic.py

```diff
@@ -66,6 +66,130 @@ def forward(self, x: Tensor, index: Optional[Tensor] = None,
+        # we could also expand `ptr` to an index
+        self.assert_index_present(index)
+
+        if dim_size is None:
+            dim_size = index.max() + 1
+
+        # compute the quantile points
+        count = torch.bincount(index, minlength=dim_size)
+        cumsum = torch.cumsum(count, dim=0)
+        q_point = self.q * (count - 1) + cumsum - count
+
+        # shape used for expansions
+        # (1, ..., 1, -1, 1, ..., 1)
+        shape = [1] * x.dim()
+        shape[dim] = -1
+
+        # two sorts: the first one on the value,
+        # the second (stable) on the indices
+        x, x_perm = torch.sort(x, dim=dim)
+        index = index.view(*shape).expand_as(x)
+        index = index.take_along_dim(x_perm, dim=dim)
+        index, index_perm = torch.sort(index, dim=dim, stable=True)
+        x = x.take_along_dim(index_perm, dim=dim)
+
+        # compute the quantile interpolations
+        if self.interpolation == 'lower':
+            idx = torch.floor(q_point).long()
+            median = x.index_select(dim, idx)
+        elif self.interpolation == 'higher':
+            idx = torch.ceil(q_point).long()
+            median = x.index_select(dim, idx)
+        elif self.interpolation == 'nearest':
+            idx = torch.round(q_point).long()
+            median = x.index_select(dim, idx)
+        else:
+            l_idx = torch.floor(q_point).long()
+            r_idx = torch.ceil(q_point).long()
+            l_med = x.index_select(dim, l_idx)
+            r_med = x.index_select(dim, r_idx)
+
+            if self.interpolation == 'linear':
+                q_frac = torch.frac(q_point).view(*shape)
+                median = l_med + (r_med - l_med) * q_frac
+            else:  # 'midpoint'
+                median = 0.5 * l_med + 0.5 * r_med
+
+        # if the number of elements is 0, return 'nan'/fill_value
+        out_mask = (count > 0).view(*shape)
+        return torch.where(
+            out_mask, median,
+            torch.tensor([self.fill_value], dtype=median.dtype,
+                         device=median.device))
+
+    def __repr__(self) -> str:
+        return (f'{self.__class__.__name__}(q={self.q}), '
+                f'interpolation={self.interpolation}, '
+                f'fill_value={self.fill_value})')
+
+
+class MedianAggregation(QuantileAggregation):
+    r"""Median aggregation operator, as used in `"Understanding Structural
+    Vulnerability in Graph Convolutional Networks"
+    <https://www.ijcai.org/proceedings/2021/310>`_.
+
+    .. note::
+        If the median lies between two values, the lowest one is returned.
+        To compute the midpoint (or other kind of interpolation) of the two
+        values, use :class:`QuantileAggregation` instead.
+
+    Args:
+        fill_value (float, optional): Default value in the case no entry is
+            found for a given index (default: :obj:`NaN`).
+    """
+    def __init__(self, fill_value: float = float('nan')):
```

Comment:
why not include interpolation as an option here too?

Reply:
I followed the [`median` implementation](https://pytorch.org/docs/stable/generated/torch.median.html) in PyTorch (also they suggest to use `quantile` for different interpolations). Also, in the future `median` could also return the index (if this is implemented in torch-scatter), and this can be done only for `"lower"`, `"higher"`, and `"nearest"` interpolations. This is merely a desing choice, so I'll stick to your preferences!

Reply:
That's a good point. Then it makes sense to me.